### ISSUE TITLE
Caching lambda expressions used to manipulate objects using reflection

### ISSUE THREAD

**Issue by ekolis** [dotnet/coreclr#8105]
Sorry if this is the wrong place to suggest this; I've never tried to contribute to .NET in this way before!

Anyway, in a project of mine I wrote a serializer that relied heavily on reflection, and it was horribly slow. However I was able to speed it up quite a bit by creating, compiling, and caching a lambda expression for each property getter and setter of each class I was serializing. I was thinking something like this could be built into the CLR, to make reflection much faster. I don't know what this does to memory usage, but perhaps at least there could be a static flag to set on some class somewhere, or a core library which does what I was doing in case anyone needs it?

**Comment by mattwarren** [comment]
I agree this would be nice to have, at the moment you have to use `IL.Emit` or `Expression` trees to speed things up, which aren't straight forward (although libraries like [Sigil](https://github.com/kevin-montrose/Sigil) can help)

It's worth pointing out that Java has this feature, it's called ['inflation'](http://stackoverflow.com/questions/10082523/java-what-is-jitcs-reflection-inflation), however it can cause [memory issues](http://spyglasstools.com/support/blog/understanding-java-inflation/)

**Comment by madelson** [comment]
According to the link @mattwarren posted, in Java the memory issues stem from having to generate a whole new class each time reflection on a method is optimized. In .NET where we have support for delegates, it seems like the overhead could potentially be lower.
